blaze-core/src/main/scala/org/http4s/blaze/util/ProcessWriter.scala

```diff
@@ -52,7 +53,7 @@ trait ProcessWriter {
   final private def go(p: Process[Task, ByteVector], stack: List[StackElem], cb: CBType): Unit = p match {
     case Emit(seq) if seq.isEmpty =>
       if (stack.isEmpty) writeEnd(ByteVector.empty).onComplete(completionListener(_, cb))
-      else go(Try(stack.head.apply(End).run), stack.tail, cb)
+      else Trampoline(go(Try(stack.head.apply(End).run), stack.tail, cb))
```

Comment:
This is not a data structure, its a inline function defined [here](https://github.com/http4s/http4s/pull/401/files#diff-dba39bf311e9e4210b06bf93b887a6f4R105). The function forks the call by name contents to a thread local `ExecutionContext` that stores all computations it receives in a queue, running all of them until its empty.